cobalt/browser/metrics/cobalt_metrics_service_client.cc

```diff
@@ -0,0 +1,213 @@
+  // reference to it.
+  auto pref_registry = base::MakeRefCounted<PrefRegistrySimple>();
+  metrics::MetricsService::RegisterPrefs(pref_registry.get());
+
+  PrefServiceFactory pref_service_factory;
+  pref_service_factory.set_user_prefs(
+      base::MakeRefCounted<InMemoryPrefStore>());
+
+  pref_service_ = pref_service_factory.Create(std::move(pref_registry));
+  // `local_state` is a common alias used in Content embedders.
+  auto* local_state = pref_service_.get();
+
+  base::FilePath user_data_dir;
+  // TODO(b/372559349): use a real path.
+  base::PathService::Get(content::PATH_START, &user_data_dir);
+
+  metrics_state_manager_ = metrics::MetricsStateManager::Create(
+      local_state, /* enabled_state_provider= */ this,
+      /* backup_registry_key= */ std::wstring(), user_data_dir
+      // Other params left as by-default.
+  );
+
+  metrics_service_ = std::make_unique<metrics::MetricsService>(
+      metrics_state_manager_.get(), this, pref_service_.get());
+  metrics_service_->InitializeMetricsRecordingState();
+}
+
+CobaltMetricsServiceClient::~CobaltMetricsServiceClient() = default;
+
+void CobaltMetricsServiceClient::Start() {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  DCHECK(IsInitialized());
+  metrics_service_->Start();
+}
+
+void CobaltMetricsServiceClient::Stop() {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  DCHECK(IsInitialized());
+  metrics_service_->Stop();
+}
+
+variations::SyntheticTrialRegistry*
+CobaltMetricsServiceClient::GetSyntheticTrialRegistry() {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  DCHECK(IsInitialized());
+  return synthetic_trial_registry_.get();
+}
+
+metrics::MetricsService* CobaltMetricsServiceClient::GetMetricsService() {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  DCHECK(IsInitialized());
+  return metrics_service_.get();
+}
+
+void CobaltMetricsServiceClient::SetMetricsClientId(
+    const std::string& client_id) {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  DCHECK(IsInitialized());
+}
+
+int32_t CobaltMetricsServiceClient::GetProduct() {
+  // TODO(b/372559349): Return Cobalt product ID.
+  return 0;
+}
+
+std::string CobaltMetricsServiceClient::GetApplicationLocale() {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  DCHECK(IsInitialized());
+  return base::i18n::GetConfiguredLocale();
+}
+
+const network_time::NetworkTimeTracker*
+CobaltMetricsServiceClient::GetNetworkTimeTracker() {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  DCHECK(IsInitialized());
+  return nullptr;
+}
+
+bool CobaltMetricsServiceClient::GetBrand(std::string* brand_code) {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  DCHECK(IsInitialized());
+  return false;
+}
+
+metrics::SystemProfileProto::Channel CobaltMetricsServiceClient::GetChannel() {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  DCHECK(IsInitialized());
+  // TODO(b/372559349): Figure out the channel, if needed.
+  return metrics::SystemProfileProto::CHANNEL_CANARY;
```

Comment:
`return ::metrics::SystemProfileProto::CHANNEL_UNKNOWN;` is likely more appropriate 

perhaps reference Cobalt's implementation for these stubbed out values to stay consistent: https://source.corp.google.com/h/github/youtube/cobalt/+/25.lts.1+:cobalt/browser/metrics/cobalt_metrics_service_client.cc;l=78-179;drc=d9fdfb83e0472aeaadb5d6ba238798021f9a2f67